.github/workflows/published-charms-tests.yaml

```diff
@@ -0,0 +1,104 @@
+name: Broad Charm Compatibility Tests
+on:
+  schedule:
+  - cron: 0 1 25 * *
+  workflow_dispatch: null
+jobs:
+  charm-tests:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout the ${{ matrix.charm-repo }} repository
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ matrix.charm-repo }}
+    - name: Checkout the operator repository
+      uses: actions/checkout@v4
+      with:
+        path: myops
+    - name: Install patch dependencies
+      run: pip install poetry~=1.6
+    - name: Update 'ops' dependency in test charm to latest
+      run: "rm -rf myops/test\nif [ -e \"test-requirements.txt\" ]; then\n  sed -i\
```

Comment:
I see your comment above, but this is going to be almost impossible to edit/maintain. Is there some way to have this in a separate file? Or have the source of it be in the update-published-charms script file in a reasonable format?

Reply:
Hmm, fair point. Maybe it is best to go back to reading and patching the file as text rather than YAML? I did look into whether I could get pyyaml to roundtrip this as it was but couldn't find any option that would do that without having to write something custom for dumping.